src/ucp/proto/proto_common.c

```diff
@@ -519,23 +519,29 @@ ucp_proto_common_find_lanes(const ucp_proto_init_params_t *params,
         }
 
         /* Check memory registration capabilities for zero-copy case */
-        if (reg_mem_type != UCS_MEMORY_TYPE_UNKNOWN) {
+        if (send_mem_type != UCS_MEMORY_TYPE_UNKNOWN) {
```

Comment:
check also for SEND_ZCOPY here, for consistency, since the name of the field is not reg_mem_type anymore. and remove the next assert in line 523

Reply:
it broke the original fix. The idea was to check for registration ability even if SEND_ZCOPY is not set (to work with mtype protocols). Revert?